### PR TITLE
HPCC-20098 Fail session unlock if modified and return modified status

### PR DESCRIPTION
**Pull request by kenrowland**
Fail an unlock session request if session modified, unless force
flag is set. Return modified status of environment for a requests
returning status.

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by kenrowland** [comment]
@rpastrana Please review

**Comment by kenrowland** [comment]
If anyone is interested, I accidentally closed the PR, so I reopened it.

File: esp/scm/ws_config2.ecm [hunk 1]
```diff
@@ -41,6 +41,7 @@ ESPresponse [exceptions_inline] EmptyResponse
 
 ESPresponse StatusResponse
 {
+    bool EnvironmentModified;
```
Comment (rpastrana) [inline esp/scm/ws_config2.ecm]: this should be part of statustype struct, unless statustype is really an "errors" structure

File: esp/scm/ws_config2.ecm [hunk 2]
```diff
@@ -329,6 +330,7 @@ ESPrequest UnlockSessionRequest
 {
     string SessionId;
     string SessionLockKey;
+    bool   FailIfModified(true);
```
Comment (rpastrana) [inline esp/scm/ws_config2.ecm]: perhaps too picky, but fail seems to imply something greater than rejecting the request...
how about rejectIfModified?

**Review by rpastrana — COMMENTED** [review]
@kenrowland couple of comments

File: esp/scm/ws_config2.ecm (hunk 1 above)
Comment (kenrowland) [reply esp/scm/ws_config2.ecm]: StatusType is just the status array. It is reused in other response structures.

File: esp/scm/ws_config2.ecm (hunk 2 above)
Comment (kenrowland) [reply esp/scm/ws_config2.ecm]: Changed.

**Review by kenrowland — COMMENTED** [review]
@rpastrana Please review latest commit.

**Review by rpastrana — APPROVED** [review]
@kenrowland

**Comment by kenrowland** [comment]
@richardkchapman please merge

**Pull request merged**